lib/Db/CollectedAddressMapper.php

```diff
@@ -40,25 +40,34 @@ public function __construct(IDBConnection $db) {
 	 *
 	 * @param string $userId
 	 * @param string $query
+	 *
 	 * @return CollectedAddress[]
 	 */
 	public function findMatching($userId, $query) {
-		$sql = 'SELECT * FROM *PREFIX*mail_collected_addresses WHERE `user_id` = ? AND (`email` ILIKE ? OR `display_name` ILIKE ?)';
-		$params = [
-			$userId,
-			'%' . $query . '%',
-			'%' . $query . '%',
-		];
-		return $this->findEntities($sql, $params);
+		$qb = $this->db->getQueryBuilder();
+		$dbQuery = $qb
+			->select('*')
+			->from($this->getTableName())
+			->where(
+				$qb->expr()->eq('user_id', $qb->createNamedParameter($userId)),
+				$qb->expr()->orX(
```

Comment:
Below you do `andWhere` on the sexond expression. Might make sense to do it consitently one way. (I prefer the andWhere since it makes it explicit it is an AND

Reply:
To be clear I mean

```
->where(
  <1>,
  <2>);
```

vs

```
->where(<1>)
->andWhere(<2>);
```